### ISSUE TITLE
Missing Characters

### ISSUE THREAD

**Issue by gregschein** [zombieFox/awesomeSheet#115]
Since the last update, my characters are no longer available, being replaced with the 2 default characters.

**Comment by zombieFox** [comment]
I'm sorry to hear that, it may be a bug. Do you know if your browser cache was cleared? 

Have you ever backed up your characters with the Export function?

**Comment by gregschein** [comment]
Backed up some, unfortunately they players didnt.
Interestingly, when I open it on my phone, it shows the character for a moment, then switches back to the demo characters.

**Comment by zombieFox** [comment]
That's unfortunate. I'm investigating the problem now. I can't seem to recreate the issue. Can you send me a copy of your backed up (JSON files) characters please? My email should be on my profile. Or attach them to a post here?

**Comment by gregschein** [comment]
Just sent by email.
Yeah pretty strange bug. We can recreate them pretty easily, so aside from wealth its not that big an issue.

**Comment by zombieFox** [comment]
Thank you, received. I'm using it to try and recreate the issue. 

I've also pushed out a bug fix to address an issue with the recent update to the Event (xp and wealth) logging. Can you refresh awesomeSheet (but don't clear your cache as your character data may still be saved there) and see if your characters still don't load?

**Comment by zombieFox** [comment]
If you're familiar with Chrome and it's Dev Tools, there is a Console command you can run to copy all character data in local storage to your clipboard: `copy(sheet.getAllCharacters())`.
[(this is what that command does)](https://github.com/zombieFox/awesomeSheet/blob/a63eacbb9b612927ae56187d6098504935a373dc/src/js/sheet.js#L27)

**Comment by gregschein** [comment]
I'll try it in but a moment. Currently GMing as we speak. 
Upon refreshing it only shows the default characters.

**Comment by zombieFox** [comment]
Understood, when you get time, GMing comes first. 
Unfortunately, I've not yet been able to recreate the issue. I'll keep at it.

**Comment by zombieFox** [comment]
I think I got it. I recently moved the app over to HTTPS and from last week started redirecting user to the new url (https://zombiefox.github.io/awesomeSheet/). Up until then some users were using the insecure address (HTTP [no **s**]) and their character data is still there. I've released a fix to disable redirecting. 

To recover your character data from the HTTP site follow these steps:

1. Enter this address into your browser address bar (take note that there is no **s** in the HTTP): http://zombiefox.github.io/awesomeSheet/
2. Hit enter and then hit ESC to stop the browser from redirecting to HTTPS.
3. If awesomeSheet is loaded and you're still on HTTP [no **s**] then hold shift and hit Refresh. This will update that instance of awesomeSheet with the fix to stop redirecting.

Your character data should be there. I'm sorry for this unforeseen bug. I'll look into this problem some more and will try to address it in the future.
